src/integrations/prefect-redis/prefect_redis/locking.py

```diff
@@ -112,20 +147,43 @@ async def aacquire_lock(
         acquire_timeout: Optional[float] = None,
         hold_timeout: Optional[float] = None,
     ) -> bool:
+        """
+        Acquires a lock asynchronously.
+
+        Args:
+            key: Unique identifier for the transaction record.
+            holder: Unique identifier for the holder of the lock. Must match the
+                holder provided when acquiring the lock.
+            acquire_timeout: Maximum time to wait for the lock to be acquired.
+            hold_timeout: Maximum time to hold the lock.
+
+        Returns:
+            True if the lock was acquired, False otherwise.
+        """
         lock_name = self._lock_name_for_key(key)
         lock = self._locks.get(lock_name)
-        if lock is not None and self.is_lock_holder(key, holder):
-            return True
-        else:
-            lock = AsyncLock(
+
+        if lock is not None and isinstance(lock, AsyncLock):
+            if await lock.owned() and lock.local.token == holder.encode():
+                return True
+            else:
+                lock = None
+
+        # Handles the case where a lock might have been released during a task retry
+        # If the lock doesn't exist in Redis at all, this method will succeed even if
+        # the holder ID doesn't match the original holder.
+        if lock is None:
+            new_lock = AsyncLock(
                 self.async_client, lock_name, timeout=hold_timeout, thread_local=False
             )
-        lock_acquired = await lock.acquire(
-            token=holder, blocking_timeout=acquire_timeout
-        )
-        if lock_acquired:
-            self._locks[lock_name] = lock
-        return lock_acquired
+            lock_acquired = await new_lock.acquire(
+                token=holder, blocking_timeout=acquire_timeout
+            )
+            if lock_acquired:
+                self._locks[lock_name] = new_lock
+            return lock_acquired
+
+        return False
```

Comment:
Why did you update this method to no longer use `is_lock_holder`? Is an update to `acquire` also necessary to have equivalent implementations?

Reply:
i was thinking we could use `owned` which seemed helpful but shouldve updated the other method yeah, happy to chat on that for a later PR, reverted to keep this diff about the pickling stuff 👍